PSICover/PSICover.cs

```diff
@@ -146,8 +147,79 @@ class Analyzer {
       ExecProgram ($"{Dir}/{RunExe}", "");
    }
 
+   double GetFileCoverage (string file) {
+      var blocks = mBlocks.Where (a => a.File == file);
+      ulong[] hits = File.ReadAllLines ($"{Dir}/hits.txt").Select (ulong.Parse).ToArray ();
+      var p = blocks.Where (a => hits[a.Id] > 0).Count () * 100.0 / blocks.Count ();
+      return Math.Round (p, 1);
+   }
+
+   void GenTable () {
+      ulong[] hits = File.ReadAllLines ($"{Dir}/hits.txt").Select (ulong.Parse).ToArray ();
+      int cBlocks = mBlocks.Count, cHit = hits.Count (a => a > 0);
+      double percent = Math.Round (100.0 * cHit / cBlocks, 1);
+
+      string text = $$"""
+            <h2>PSI Tests
+            Total coverage: {{percent}}%</h2>
+            <table>
+               <tr>
+               <th>Module</th>
+               <th>File</th>
+               <th>Total blocks</th>
+               <th>Hit blocks</th>
+               <th>Coverage</th>
+               </tr>
+            """;
+      var modules = mBlocks.Select (a => a.Module).Distinct ().ToArray ();
+      foreach (var module in modules) {
+         text += $$"""
+               <tr>
+               <td>{{char.ToUpper (module[0]) + module[1..]}}</td>
+               <td></td>
+               <td></td>
+               </tr>
+            """;
+         var files = mBlocks.Where (a => a.Module == module).Select (a => a.File).Distinct ().ToArray ();
+         List<(string file, int tBlocks, int hBlocks, double coverage)> fTable = new ();
+         foreach (var file in files) {
+            var fBlocks = mBlocks.Where (a => a.File == file);
+            int hBlocks = fBlocks.Where (a => hits[a.Id] > 0).Count ();
+            fTable.Add ((file, fBlocks.Count (), hBlocks, GetFileCoverage (file)));
+         }

```

Comment:
`fBlocks.Count()` will go through the mBlocks list to get the total count. Similarly, in `GetFileCoverage (file)`, you are going through the mBlocks list again to determine the percent. You could just pass `fBlocks.Count ()` and hBlocks to the function. Better do the percent calculation here itself.

Reply:
Changed helper method